### PR TITLE
Implement file duplication

### PR DESCRIPTION
**Pull request by alvra**
This PR implements duplicating files in the fs tree context menu.

### CODE REVIEW

**Comment by panekj** [comment]
I don't think adding duplication is a proper thing to do, I'd rather see `Copy`/`Paste`/`Cut` actions like in standard file explorers

**Comment by alvra** [comment]
What would be the difference between file duplication and copy? Do you mean copy to clipboard?

**Comment by panekj** [comment]
> What would be the difference between file duplication and copy? 

Duplication implements very specific action which can be replaced by clipboard actions

> Do you mean copy to clipboard?

Yes

**Comment by dzhou121** [comment]
I personally feel it's a good addition to copy/paste/cut, because I often do copy and paste which is essentially duplicate.

**Comment by panekj** [comment]
> I personally feel it's a good addition to copy/paste/cut, because I often do copy and paste which is essentially duplicate.

It's not bad per se, but I would like to avoid adding it to mouse context menu. As we develop we will be adding more actions which could be more useful and cluttering context menu with too many actions feels wrong.

**Comment by alvra** [comment]
True, duplication is more specific than copy+paste. Personally I use duplicate much more often and having it as a single option is helpful to me. New files often share code with its siblings so I probably use duplicate more even than "new empty file".

**Comment by dzhou121** [comment]
> > I personally feel it's a good addition to copy/paste/cut, because I often do copy and paste which is essentially duplicate.
> 
> It's not bad per se, but I would like to avoid adding it to mouse context menu. As we develop we will be adding more actions which could be more useful and cluttering context menu with too many actions feels wrong.

I don't feel it would be too much with one more menu item in the copy/paste/cut section(in the future).

**Comment by panekj** [comment]
> New files often share code with its siblings so I probably use duplicate more even than "new empty file".

We should probably hide `New ...` when targeting files like VSCode and show it on directories/empty space only

**Comment by dzhou121** [comment]
> > New files often share code with its siblings so I probably use duplicate more even than "new empty file".
> 
> We should probably hide `New ...` when targeting files like VSCode and show it on directories/empty space only image image

Yeah in a separate PR maybe.

**Comment by panekj** [comment]
> > > I personally feel it's a good addition to copy/paste/cut, because I often do copy and paste which is essentially duplicate.
> > 
> > 
> > It's not bad per se, but I would like to avoid adding it to mouse context menu. As we develop we will be adding more actions which could be more useful and cluttering context menu with too many actions feels wrong.
> 
> I don't feel it would be too much with one more menu item in the copy/paste/cut section(in the future).

If we're not going to implement custom context actions from plugins, then yes, it should stay fine but that's quite a limitation IMO

File: lapce-data/src/explorer.rs [hunk 1]
```diff
@@ -38,12 +39,25 @@ pub enum Naming {
...
+    Duplicating {
+        /// The index that the file being created should appear at
+        /// Note that when duplicating, it is not yet actually created.
+        list_index: usize,
+        /// Indentation level
+        indent_level: usize,
+        /// The folder that the file/directory is being created within
+        base_path: PathBuf,
+        /// The name of the file being duplicated
+        name: OsString,
```
Comment (panekj) [inline lapce-data/src/explorer.rs:52]: This should be a `String` rather than `OsString`, we don't operate on raw platform strings (at least not yet)

**Comment by dzhou121** [comment]
Can you please resolve the CHANGELOG.md conflict?

**Pull request merged**